src/wmtk/MultiMeshManager.cpp

```diff
@@ -0,0 +1,145 @@
+        // update on child_mesh
+        child_mesh->multi_mesh_manager.map_to_parent_handle = map_to_parent_handle;
+        child_mesh->multi_mesh_manager.m_is_parent_mesh = false;
+        child_mesh->multi_mesh_manager.m_child_id = long(parent_mesh.multi_mesh_manager.child_meshes.size());
+
+        // update on parent_mesh
+        parent_mesh.multi_mesh_manager.child_meshes.push_back(child_mesh);
+        parent_mesh.multi_mesh_manager.map_to_child_handles.push_back(map_to_child_handle);
+        parent_mesh.multi_mesh_manager.m_is_parent_mesh = true;
+       
+    }
+
+    Tuple MultiMeshManager::map_tuple_between_meshes(const Mesh& source_mesh, const Mesh& target_mesh, MeshAttributeHandle<long> source_map_handle, const Tuple& source_tuple)
+    {
+        PrimitiveType source_mesh_ptype = source_mesh.top_simplex_type();
+        PrimitiveType target_mesh_ptype = target_mesh.top_simplex_type();
+
+        auto [source_mesh_base_tuple, target_mesh_base_tuple] = read_tuple_map_attribute(source_map_handle, source_mesh, source_tuple);
+
+        if (source_mesh_base_tuple.is_null_tuple())
+        {
+            return Tuple(); // return null tuple
+        }
+        
+        if (source_mesh_ptype == PrimitiveType::Face && target_mesh_ptype == PrimitiveType::Face)
+        {
+            Tuple cur_tuple = source_tuple;
+            std::vector<char> record_switch_operations;
+            while (cur_tuple != source_mesh_base_tuple)
+            {
+                cur_tuple = source_mesh.switch_tuple(cur_tuple, PrimitiveType::Vertex);
+                record_switch_operations.push_back('v');
+                if (cur_tuple == source_mesh_base_tuple)
+                {
+                    break;
+                }
+                cur_tuple = source_mesh.switch_tuple(cur_tuple, PrimitiveType::Edge);
+                record_switch_operations.push_back('e');
+            }
+
+            Tuple ret_tuple = target_mesh_base_tuple;
+            for (int i = record_switch_operations.size() - 1; i >= 0; --i)
+            {
+                if (record_switch_operations[i] == 'v')
+                {
+                    ret_tuple = target_mesh.switch_tuple(ret_tuple, PrimitiveType::Vertex);
+                }
+                else // record_switch_operations[i] == 'e'
+                {
+                    ret_tuple = target_mesh.switch_tuple(ret_tuple, PrimitiveType::Edge);
+                }
+            }
+
+            return ret_tuple;
+        }
+        else
+        {
+            // TODO: implement this for other cases later
+            // TODO: maybe use a seperate function for each case?
+            return Tuple();
+        }
+    }
+
+    bool MultiMeshManager::is_child_mesh_valid(const Mesh& parent_mesh, const Mesh& child_mesh)
+    {
+        // TODO: implement this
+        
+        return true;
+    }
+
+    std::vector<Tuple> MultiMeshManager::map_edge_tuple_to_all_children(const Mesh& parent_mesh, const Tuple& edge_tuple)
```

Comment:
i don't understand this function's implementation. It return sa collection of tuples from all of the children - so if you have two child meshes hten you just get a bunch of tuples from either mesh without any identification wher ehtey came from? Perhaps you want to specify a specific child mesh as an argument here?

Reply:
oh. i misunderstood what this function does - it assumes that `child_meshes[i]->child_id() == i` which i guess is true right now and returns every copy of a vector if it exists.

i think that constructing this vector like this is not necessary - but i will comment on it in your Executor implementations of operations because the structure isn't necessary